### PR TITLE
D3D: Implement vectored efb pokes

### PR DESCRIPTION
**Pull request by stenzek**
Implementing the vectored EFB poke path for the D3D backend that is currently present in the GL backend.

This needs testing, I haven't identified any regressions, but if there is anyone who has titles that heavily use EFB pokes, could you please confirm this does not change anything, visually. Performance should be on par with GL now.

Additionally, I've increased the util vertex buffer to 64KiB, to reduce the number of draws which have to be issued to draw an entire frame with efb pokes. That buffer class is rather iffy though, and I had to modify it in order to not have to have an intermediate buffer, probably would be an idea in the future to create a general "streaming" buffer (and perhaps share with the vertex loader).

### CODE REVIEW

File: Source/Core/VideoBackends/D3D/Render.cpp [hunk 1]
```diff
@@ -474,56 +474,54 @@ u32 Renderer::AccessEFB(EFBAccessType type, u32 x, u32 y, u32 poke_data)
 		else if (alpha_read_mode.ReadMode == 1) return (ret | 0xFF000000); // GX_READ_FF
 		else /*if(alpha_read_mode.ReadMode == 0)*/ return (ret & 0x00FFFFFF); // GX_READ_00
 	}
-	else if (type == POKE_COLOR)
+	else // if (type == POKE_COLOR || type == POKE_Z)
 	{
-		u32 rgbaColor = (poke_data & 0xFF00FF00) | ((poke_data >> 16) & 0xFF) | ((poke_data << 16) & 0xFF0000);
+		EfbPokeData d;
```
Comment (degasus) [inline Source/Core/VideoBackends/D3D/Render.cpp]: I think this kind of fallback should be within the caller. So it could be removed from OGL too.

**Comment by degasus** [comment]
OGL + VideoCommon LGTM, no clue about D3D

**Comment by degasus** [comment]
@hdcmeta Do you want to review the d3d11 parts?

**Comment by skidau** [comment]
Tested this out.  It works fine and speeds up EFB pokes by around 7 times.  Tested Avatar: The Last Airbender (GC), Jimmy Neutron: Boy Genius (GC) and Mario Strikers Charged (Wii).

**Comment by LuismaSP89** [comment]
Oh my god, very very great improvement : ) works awesome on Super Mario Galaxy, Resident evil 4 GC, The legend of Zelda Windwaker, etc. No problems found.

**Pull request merged**